src/merkle/merkle_tree.rs

```diff
@@ -114,6 +114,28 @@ impl MerkleTree {
         Ok(path.into())
     }
 
+    // ITERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an iterator over the leaves of this [MerkleTree].
+    pub fn leaves(&self) -> core::slice::Iter<Word> {
+        let leaves_start = self.nodes.len() / 2;
+        self.nodes[leaves_start..].iter()
+    }
+
+    /// Returns n iterator over every inner node of this [MerkleTree].
+    ///
+    /// The iterator order is unspecified.
+    pub fn inner_nodes(&self) -> InnerNodeIterator<'_> {
```

Comment:
I don't think we need to specify the lifetime here as the compiler should be able to infer it from `&self`

Reply:
Good catch! Fixed!